fm2prof/MaskOutputFile.py

```diff
@@ -20,68 +20,72 @@
 Stichting Deltares and remain full property of Stichting Deltares at all times.
 All rights reserved.
 """
-
-import os
-
+from pathlib import Path
+from typing import Union, Optional
 import geojson
 
 
+
 class MaskOutputFile:
     @staticmethod
-    def create_mask_point(coords: geojson.coords, properties: dict):
+    def create_mask_point(coords: geojson.coords, properties: Optional[dict] = None) -> geojson.Feature:
         """Creates a Point based on the properties and coordinates given.
 
         Arguments:
             coords {geojson.coords} --
                 Coordinates tuple (x,y) for the mask point.
             properties {dict} -- Dictionary of properties
         """
-        output_mask = geojson.Feature(geometry=geojson.Point(coords))
+        if not coords:
+            raise ValueError("coords cannot be empty.")
+        output_mask = geojson.Feature(geometry=geojson.Point(coords))       
+        
         if properties:
             output_mask.properties = properties
         return output_mask
 
     @staticmethod
-    def validate_extension(file_path: str):
-        if not file_path:
-            # Should not be evaluated
-            return
-        if not file_path.endswith(".json") and not file_path.endswith(".geojson"):
+    def validate_extension(file_path: Union[str, Path]) -> None:
+        if not isinstance(file_path, (str, Path)):
+            err_msg = f"file_path should be string or Path, not {type(file_path)}"
+            raise TypeError(err_msg)
+        if Path(file_path).suffix not in [".json", ".geojson"]:
             raise IOError(
-                "Invalid file path extension, " + "should be .json or .geojson."
+                "Invalid file path extension, should be .json or .geojson."
             )
 
     @staticmethod
-    def read_mask_output_file(file_path: str):
+    def read_mask_output_file(file_path: Union[str, Path]) -> dict:
         """Imports a GeoJson from a given json file path.
 
         Arguments:
             file_path {str} -- Location of the json file
         """
-        geojson_data = geojson.FeatureCollection(None)
-        if not file_path or not os.path.exists(file_path):
-            return geojson_data
+        file_path = Path(file_path)
+        if not file_path.exists():
+            err_msg = f"File path {file_path} not found."
+            raise FileNotFoundError(err_msg)
 
         MaskOutputFile.validate_extension(file_path)
-        with open(file_path) as geojson_file:
-            try:
-                geojson_data = geojson.load(geojson_file)
-            except:
-                return geojson_data
-
+        with file_path.open("r") as geojson_file:            
+            geojson_data = geojson.load(geojson_file)
+        if not isinstance(geojson_data, geojson.FeatureCollection):

```

Comment:
Is this check actually needed? Can `geojson_data` be something else than a `geojson.FeatureCollection`?

Reply:
Yes, geojson will load a json as a dict. If the file is a geojson it will return a geojson.FeatureCollection